--- qt-everywhere-commercial-src-4.8.2/src/3rdparty/webkit/Source/WebCore/page/qt/EventHandlerQt.cpp
+++ qt-everywhere-commercial-src-4.8.2/src/3rdparty/webkit/Source/WebCore/page/qt/EventHandlerQt.cpp
@@ -105,6 +105,8 @@ PassRefPtr<Clipboard> EventHandler::createDraggingClipboard() const

 bool EventHandler::passMousePressEventToSubframe(MouseEventWithHitTestResults& mev, Frame* subframe)
 {
+    if (subframe->page())
+        subframe->page()->focusController()->setFocusedFrame(subframe);
     subframe->eventHandler()->handleMousePressEvent(mev.event());
     return true;
 }
