--- qt-everywhere-commercial-src-4.8.4/src/gui/painting/qpdf.cpp	2012-11-23 11:28:53.000000000 +0100
+++ qt-everywhere-commercial-new-4.8.4/src/gui/painting/qpdf.cpp	2012-12-06 11:26:26.894333464 +0100
@@ -1053,7 +1053,16 @@
 
     if (d->simplePen) {
         // draw strokes natively in this case for better output
+        QBrush oldBrush = d->brush;
+        if (d->opacity != 1.0 && !d->hasBrush) {
+            d->brush = d->pen.brush();
+            setBrush();
+        }
         *d->currentPage << QPdf::generatePath(p, QTransform(), d->hasBrush ? QPdf::FillAndStrokePath : QPdf::StrokePath);
+        if (oldBrush != d->brush) {
+            d->brush = oldBrush;
+            setBrush();
+        }
     } else {
         if (d->hasBrush)
             *d->currentPage << QPdf::generatePath(p, d->stroker.matrix, QPdf::FillPath);
